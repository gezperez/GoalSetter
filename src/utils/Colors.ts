const Colors = {
  LOADER_BACKGROUD: 'rgba(255, 255, 255, 0.9)',
  HEADER_GRADIENT_START: '#8D24C4',
  HEADER_GRADIENT_END: '#500F71',
  WHITE: '#FFFFFF',
  BORDER: '#D2D2D2',
  FOCUSED_BORDER: '#00B7D5',
  LABEL: '#D8D8D8',
  ERROR: '#FF0000',
  BUTTON: '#5FBA20',
  BLACK: '#000000',
  LINK: '#17BFDD',
  HOME_TITLE: '#3B2F71',
}

export default Colors
