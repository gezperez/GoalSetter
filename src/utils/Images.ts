const Image = {
  EMAIL: require('~/assets/images/email.png'),
  LOCK: require('~/assets/images/lock.png'),
  PERSON: require('~/assets/images/person.png'),
  MONEY_BACKGROUND: require('~/assets/images/money-background.png'),
  FLYING_MONEY: require('~/assets/images/flying-money.png'),
  HEADER_BACKGROUND: require('~/assets/images/background.png'),
  HOME_ICON: require('~/assets/images/home-icon.png'),
}

export default {
  EMAIL: Image.EMAIL,
  LOCK: Image.LOCK,
  PERSON: Image.PERSON,
  MONEY_BACKGROUND: Image.MONEY_BACKGROUND,
  FLYING_MONEY: Image.FLYING_MONEY,
  HEADER_BACKGROUND: Image.HEADER_BACKGROUND,
  HOME_ICON: Image.HOME_ICON,
}
