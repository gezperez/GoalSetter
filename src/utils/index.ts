export { default as Colors } from './Colors'
export { default as Typographys } from './Typographys'
