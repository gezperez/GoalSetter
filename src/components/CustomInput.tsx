import React, { forwardRef, useCallback, useState } from 'react'
import {
  Animated,
  Image,
  ImageSourcePropType,
  ImageStyle,
  StyleSheet,
  TextInput,
  TextInputProps,
  TextStyle,
  View,
  ViewStyle,
} from 'react-native'
import { size } from 'lodash'

import { useTween } from '~/hooks'
import { Colors, Typographys } from '~/utils'

export type CustomInputProps = {
  icon: ImageSourcePropType
  label: string
  error?: string
} & TextInputProps

const CustomInput = forwardRef<TextInput, CustomInputProps>(
  ({ icon, label, error, accessibilityLabel, testID, onChangeText, value, ...props }, ref) => {
    const [hasFocus, setHasFocus] = useState(false)

    const animateLabel = hasFocus || size(value) >= 1

    const [interpolate] = useTween(animateLabel, 200)

    const handleBlur = useCallback(() => setHasFocus(false), [])

    const handleFocus = useCallback(() => setHasFocus(true), [])

    return (
      <Animated.View
        style={[
          styles.container,
          {
            borderBottomColor: hasFocus ? Colors.FOCUSED_BORDER : Colors.BORDER,
          },
          error ? styles.error : {},
        ]}
      >
        <Image source={icon} style={styles.image} />

        <View style={styles.inputContainer}>
          <Animated.Text
            numberOfLines={1}
            style={[
              styles.label,
              {
                top: interpolate(0, -28),
                fontSize: interpolate(17, 12),
                lineHeight: interpolate(24, 14),
              },
            ]}
          >
            {label}
          </Animated.Text>
          <TextInput
            ref={ref}
            style={styles.input}
            value={value}
            onBlur={handleBlur}
            onFocus={handleFocus}
            testID={testID}
            accessibilityLabel={accessibilityLabel}
            onChangeText={onChangeText}
            {...props}
          />
        </View>
      </Animated.View>
    )
  }
)

type Styles = {
  container: ViewStyle
  inputContainer: ViewStyle
  input: TextStyle
  label: TextStyle
  error: ViewStyle
  image: ImageStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottomWidth: 0.5,
    borderBottomColor: Colors.BORDER,
    height: 50,
    marginVertical: 16,
    marginHorizontal: 36,
  },
  label: {
    ...Typographys.label,
  },
  error: {
    borderBottomColor: Colors.ERROR,
  },
  inputContainer: {
    flex: 1,
    justifyContent: 'center',
  },
  input: {
    ...Typographys.input,
    position: 'absolute',
    left: 0,
    right: 0,
  },
  image: {
    resizeMode: 'contain',
    width: 16,
    height: 16,
    marginRight: 16,
  },
})

export default CustomInput
