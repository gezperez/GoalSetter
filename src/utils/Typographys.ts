import { StyleSheet, TextStyle } from 'react-native'

import Colors from './Colors'

type Typographys = {
  headerTitle: TextStyle
  headerSubtitle: TextStyle
  label: TextStyle
  input: TextStyle
  button: TextStyle
  text: TextStyle
  footer: TextStyle
  footerLink: TextStyle
}

const typographyObject: Typographys = {
  headerTitle: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 20,
    lineHeight: 23,
    color: Colors.WHITE,
  },
  headerSubtitle: {
    fontFamily: 'Montserrat-Light',
    fontSize: 14,
    lineHeight: 16,
    color: Colors.WHITE,
  },
  label: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 19,
    lineHeight: 16,
    color: Colors.LABEL,
  },
  input: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 19,
    lineHeight: 22,
    color: Colors.FOCUSED_BORDER,
  },
  button: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 12,
    lineHeight: 14,
    color: Colors.WHITE,
  },
  text: {
    fontFamily: 'Montserrat-Light',
    fontSize: 14,
    lineHeight: 16,
    color: Colors.BLACK,
  },
  footer: {
    fontFamily: 'Montserrat-Light',
    fontSize: 14,
    lineHeight: 16,
    color: Colors.BLACK,
    textAlign: 'center',
  },
  footerLink: {
    fontFamily: 'Montserrat-Bold',
    fontSize: 14,
    lineHeight: 16,
    color: Colors.LINK,
    textAlign: 'center',
  },
}

const Typography = StyleSheet.create<Typographys>(typographyObject)

export default Typography
