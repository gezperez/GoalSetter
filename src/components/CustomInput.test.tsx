import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import { Images } from '~/utils'
import CustomInput from './CustomInput'

describe('CustomInput', () => {
  afterEach(jest.resetAllMocks)

  it('should render with required props', () => {
    const mockOnChangeText = jest.fn()

    const { toJSON, getByTestId } = render(
      <CustomInput
        label="input"
        icon={Images.EMAIL}
        accessibilityLabel="input"
        testID="input"
        onChangeText={mockOnChangeText}
      />
    )

    const input = getByTestId('input')

    expect(input).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })

  it('should call onChangeText with expected string', () => {
    const mockOnChangeText = jest.fn()

    const { getByTestId, toJSON } = render(
      <CustomInput
        label="input"
        icon={Images.EMAIL}
        accessibilityLabel="input"
        testID="input"
        onChangeText={mockOnChangeText}
      />
    )

    const input = getByTestId('input')
    fireEvent.changeText(input, 'test')

    expect(mockOnChangeText).toHaveBeenCalledWith('test')
    expect(toJSON()).toMatchSnapshot()
  })
})
