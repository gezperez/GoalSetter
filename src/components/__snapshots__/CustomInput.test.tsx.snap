// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should call onChangeText with expected string 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#D2D2D2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "marginHorizontal": 36,
      "marginVertical": 16,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/email.png",
      }
    }
    style={
      Object {
        "height": 16,
        "marginRight": 16,
        "resizeMode": "contain",
        "width": 16,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      collapsable={false}
      nativeID="animatedComponent"
      numberOfLines={1}
      style={
        Object {
          "color": "#D8D8D8",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 17,
          "lineHeight": 24,
          "top": 0,
        }
      }
    >
      input
    </Text>
    <TextInput
      accessibilityLabel="input"
      onBlur={[Function]}
      onChangeText={
        [MockFunction] {
          "calls": Array [
            Array [
              "test",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      onFocus={[Function]}
      style={
        Object {
          "color": "#00B7D5",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 19,
          "left": 0,
          "lineHeight": 22,
          "position": "absolute",
          "right": 0,
        }
      }
      testID="input"
    />
  </View>
</View>
`;

exports[`CustomInput should render with required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#D2D2D2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "marginHorizontal": 36,
      "marginVertical": 16,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/email.png",
      }
    }
    style={
      Object {
        "height": 16,
        "marginRight": 16,
        "resizeMode": "contain",
        "width": 16,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      collapsable={false}
      nativeID="animatedComponent"
      numberOfLines={1}
      style={
        Object {
          "color": "#D8D8D8",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 17,
          "lineHeight": 24,
          "top": 0,
        }
      }
    >
      input
    </Text>
    <TextInput
      accessibilityLabel="input"
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      style={
        Object {
          "color": "#00B7D5",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 19,
          "left": 0,
          "lineHeight": 22,
          "position": "absolute",
          "right": 0,
        }
      }
      testID="input"
    />
  </View>
</View>
`;
